index.js

```diff
@@ -164,17 +167,21 @@ Warehouse.prototype.makeRequest = function makeRequest(uri, pathname, options, n
       || err.message.includes('400');
   });
 
-  operation.attempt(fn => {
-    request(options, function replied(error, response, body) {
-      if (error) return fn(error);
+  operation.attempt(async fn => {
+    try {
+      const res = await fetch(options.url, options);
+      const body = await res.json();
 
-      if (response.statusCode === 404) return fn(new Error(`404 Not Found ${JSON.stringify(body)}`));
-      if (response.statusCode === 400) return fn(new Error(`400 Bad Request ${JSON.stringify(body)}`));
+      if (res.statusCode === 404) return fn(new Error(`404 Not Found ${JSON.stringify(body)}`));
+      if (res.statusCode === 400) return fn(new Error(`400 Bad Request ${JSON.stringify(body)}`));
 
-      if (response.statusCode < 200 || response.statusCode > 299) return fn(new Error(`Invalid status code ${response.statusCode} ${body ? JSON.stringify(body) : ''}`));
+      if (res.statusCode < 200 || res.statusCode > 299) return fn(new Error(`Invalid status code ${res.statusCode} ${body ? JSON.stringify(body) : ''}`));
```

Comment:
this can just be 
```suggestion
      if (!res.ok) return fn(new Error(`Invalid status code ${res.statusCode} ${body ? JSON.stringify(body) : ''}`));
```